src/audio/pipeline.c

```diff
@@ -618,11 +661,31 @@ int pipeline_trigger(struct pipeline *p, struct comp_dev *host, int cmd)
 
 static int pipeline_comp_reset(struct comp_dev *current, void *data, int dir)
 {
+	struct pipeline *p = data;
+	int stream_direction = p->source_comp->params.direction;
+	int end_type;
 	int err = 0;
 
 	tracev_pipe("pipeline_comp_reset(), current->comp.id = %u, dir = %u",
 		    current->comp.id, dir);
 
+	if (!comp_is_single_pipeline(current, p->source_comp)) {
+		/* if pipeline connected to the starting one is in improper
+		 * direction (CAPTURE towards DAI, PLAYBACK towards HOST),
+		 * stop propagation
+		 */
+		end_type = comp_get_endpoint_type(current->pipeline->sink_comp);
+		if (stream_direction == SOF_IPC_STREAM_PLAYBACK) {
```

Comment:
Please add the comment why are the conditional checks. I take that they are to cover scenarios with independent life cycles between pipelines (i.e. direction for some of pipelines is not know yet).

Reply:
comments extended to explain why these check are made